### DISCUSSION TITLE
Create a TaskMonitor that displays progress in the status bar

### DISCUSSION THREAD

**Discussion by cyberkaida** [NationalSecurityAgency/ghidra#6583] [Q&A] [answered]
I have a long running task the user can start from a context menu. I'd like to have the task monitored with a progress bar like the auto analysis, but I can't find the right way to get a TaskMonitor to appear in the status bar. I can get the crunching bits, or no notification (using a background task), but I would really like visual feedback that the background task is running. Can someone point me to the right monitor type to use from my extension?

<img width="296" alt="Screenshot 2024-05-26 at 13 18 12" src="https://github.com/NationalSecurityAgency/ghidra/assets/118712366/d0f76801-8c08-46f5-b41f-d4fdcfbf22c1">

**Comment by dragonmacher** [comment, marked as answer]
There is a method on `PluginTool` that allows you to do this:

```
public <T extends DomainObject> void executeBackgroundCommand(BackgroundCommand<T> cmd, T obj)
```

Assuming you are in a plugin, the plugin has access to the plugin tool.   The `BackgroundCommand` class should be fairly straight-forward to use.

**Reply by cyberkaida** [reply]
I will try this again. I think I had another issue causing me to block the swing thread. I'll double check I'm not making a blocking call after scheduling the background task.
